app/controllers/api/base_controller.rb

```diff
@@ -37,11 +37,18 @@ def verify_mfa_requirement
     render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
   end
 
+  def verify_api_key_gem_scope
+    return unless @api_key.rubygem && @api_key.rubygem != @rubygem
```

Comment:
I want to highlight that rubygems.org sorely needs an authorization plugin (ex: pundit). A lot of tech debt has been accrued with checks like these all over the codebase.

Reply:
I felt a little guilty writing this line of code. Maybe delegating this logic to the model with something like `def authorized_for_gem?(gem)` would be an improvement?

> I want to highlight that rubygems.org sorely needs an authorization plugin (ex: pundit). A lot of tech debt has been accrued with checks like these all over the codebase.

Would it be worth creating an issue for this? I bet someone would be interested in looking into this (myself included).

Reply:
see https://github.com/rubygems/rubygems.org/issues/2967